packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/TablesViewer.tsx

```diff
@@ -21,22 +20,19 @@ import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from ".
 import { bufferToBigInt } from "../utils/bufferToBigInt";
 import { EditableTableCell } from "./EditableTableCell";
 
-type Props = {
-  table: string | undefined;
-};
-
-export function TablesViewer({ table: selectedTable }: Props) {
-  const [sorting, setSorting] = useState<SortingState>([]);
-  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
-  const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({});
-  const [rowSelection, setRowSelection] = useState({});
-  const [globalFilter, setGlobalFilter] = useState("");
-  const [showAllColumns, setShowAllColumns] = useState(false);
+export function TablesViewer() {
+  const [selectedTableId] = useQueryState("tableId");
+  const [globalFilter, setGlobalFilter] = useQueryState("filter", parseAsString.withDefault(""));
+  const [showAllColumns, setShowAllColumns] = useQueryState("showAllColumns", parseAsBoolean.withDefault(false));
+  const [sorting, setSorting] = useQueryState(
+    "sort",
+    parseAsArrayOf(parseAsJson<ColumnSort>()).withDefault(useRef([]).current),
+  );
 
   const { data: schema } = useQuery({
-    queryKey: ["schema", { table: selectedTable }],
+    queryKey: ["schema", { table: selectedTableId }],
     queryFn: async () => {
-      const response = await fetch(`/api/schema?table=${selectedTable}`);
+      const response = await fetch(`/api/schema?table=${selectedTableId}`);
```

Comment:
would suggest getting in the habit of encoding URL params to save you some headache later

```suggestion
      const response = await fetch(`/api/schema?table=${encodeURICompoonent(selectedTableId)}`);
```
or

```suggestion
      const response = await fetch(`/api/schema?${new URLSearchParams({ table: selectedTableId })}`);
```

same as all other usages below

Reply:
That's a nice tip, will do that